src/praisonai/praisonai/gateway/server.py

```
@@ -3126,6 +3126,39 @@ def _resolve_agent_for_message(
+        if not bindings:
+            return None
+        try:
+            from praisonaiagents.gateway import resolve_route
+
+            match = resolve_route(bindings, facts)
+            binding = getattr(match, "binding", None)
+            if binding is None:
+                return None
+            tool_policy = getattr(binding, "tool_policy", None)
+            return tool_policy() if callable(tool_policy) else None
+        except Exception as exc:  # pragma: no cover - defensive
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)